calvin/runtime/south/plugins/io/button/tkinter_impl/button.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2015 Ericsson AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import Tkinter
+import thread
+import threading
+import Queue
+
+class Button():
```

Comment:
If possible try to avoid the threading and the while(1) loop, looks like tkinter integrates with twisted: https://github.com/racker/python-twisted-core/blob/master/doc/examples/tkinterdemo.py